### PR TITLE
Has many filter

### PR DESCRIPTION
**Pull request by edpaget**

### CODE REVIEW

**Comment by camallen** [comment]
Nice work @edpaget !

File: app/serializers/concerns/filter_has_many.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  module ClassMethods
+    def page(params = {}, scope = nil, context = {})
+      filters = has_many_filterable_by.map do |filter|
+        filter << params.delete(filter[1])
+      end.delete_if do |filter|
+        filter[2].nil?
+      end
+
+      scope = filters.reduce(scope || self.model_class.all) do |query, filter|
+        scope.joins(filter[0]).where(filter[0] => {id: filter[2]})
```
Comment (edpaget) [inline app/serializers/concerns/filter_has_many.rb]: Crap I just realized this should query instead of scope

Comment (camallen) [inline app/serializers/concerns/filter_has_many.rb]: Doesn't the RP serializer page method use a scope [here](https://github.com/RestPack/restpack_serializer/blob/b1f3cb62de1af8cca11870aec4fbac5da6d901c1/lib/restpack_serializer/serializable/paging.rb#L5)?

Comment (edpaget) [inline app/serializers/concerns/filter_has_many.rb]: Yes, but I'm passing the scope into the reduce block as the `query` variable. How it works now will only allow filtering by one has many param (which is as many as any of the models have).

Comment (camallen) [inline app/serializers/concerns/filter_has_many.rb]: Do you want to revert or just fix in another PR?

**Pull request merged**